### PR TITLE
HDDS-7200. Fixed inaccurate get request status code

### PR DESCRIPTION
**Pull request by xichen01**
## What changes were proposed in this pull request?
will get the content in a range( add such as: "Range: bytes=123-" at request header), The backend service needs to return this range of data, and the returned status code should be [206](https://www.rfc-editor.org/rfc/rfc7233#section-4.1)
current Ozone s3 gateway will return 200 instead of 206, this pr will be fixed it.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7200

Please replace this section with the link to the Apache JIRA)

## How was this patch tested?

Send a get request requesting range data, the return status should be 206.

### CODE REVIEW

**Comment by xichen01** [comment]
Run cat target/integration/summary.txt
org.apache.hadoop.ozone.scm.node.TestDecommissionAndMaintenance
org.apache.hadoop.ozone.scm.node.TestDecommissionAndMaintenance

File: hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java [hunk 1]
```diff
@@ -344,6 +344,8 @@ public Response get(
         };
         responseBuilder = Response
             .ok(output)
```
Comment (kaijchen) [inline hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java:346]: This line should be removed, as it equals to setting `status = OK` and `entity = output`.

```suggestion
```

**Review by kaijchen — CHANGES_REQUESTED** [review]
Thanks @xichen01 for the patch, mostly LGTM. One small improvement here.

File: hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java (hunk 1 above)
Comment (xichen01) [reply hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java]: done

**Review by ferhui — APPROVED** [review]
+1

**Review by kaijchen — APPROVED** [review]
LGTM, thanks @xichen01.

**Pull request merged**